### PR TITLE
407 training flow

### LINKED ISSUE #402: Training Launcher (Front End)
This is the training launcher page.  before a volunteer can work the phones, they need to take the training to get up to speed.   

This task is for the UI only of the launcher.  

image

- [ ] it should have re-usable mid sections for more modules as needed.
image

- [ ] it will also have a section for custom notes (only will appear if there are any notes in organization settings)

image

https://projects.invisionapp.com/d/main#/console/14746784/338201794/preview

### LINKED ISSUE #407: Training Flow (Backend)
@JamesRArroyo great work on getting the bulk of this worked out

@BradenM  all that is left is to wire this up so that when a user first comes to the phone page we have a database setting of "phone training complete". It is assumed that nobody has been trained.  The "Start Taking Calls" button would actually say "Start Training" if the flag is null or false.  Launching the rest works. with the Pr jimmy just did.  
When training is done, set the flag so we don't prompt them again (until a day when we want them to do another round of training if that happens at a future point we would just re-set their flag and they go again).

In order for a volunteer to be able to work the phones, they need to do some quick training.  This flow is design to help make that happen. 

- [ ] **Braden:**  By default, all volunteers should have a flag "phone training complete" which by default is null
- [ ]  if the phoneTrainingComplete flag is not true, then the "Start Taking Calls" button should say "Start Training" instead.  
image
- [ ] "Start Training" Should launch the Training Launcher Page #402 
- [x] Each module on the training launcher page should link to a Training Details page #404 
- [x] Each Training Details page should have 3 simple questions that you have to get correct in order to enable the "Complete quiz" button.
- [x] If you "Complete the Quiz" then the window closes and you update the training item to "Completed"  https://projects.invisionapp.com/d/main#/console/14746784/338201796/preview
- [x] for now totally skip the custom instructions thing... no time
- [x] once all training modules are "completed" (we will only have 2 for a while), then enable the "Next button". https://projects.invisionapp.com/d/main#/console/14746784/338201796/preview
- [x] The Next button will take us to the "Training Finish" page #405 which will allow the user to review and edit their phone number and languages.   
- [ ] **Braden:** Hitting "Confirm" will set the "phoneTrainingComplete" flag to true and the "Avialibilty" Button will now do what it is supposed to do.

### PR DESCRIPTION
**Pull request by JamesRArroyo**
This closed #402 and #404 
Once hooked up to the db it will close #407

- Training button now pops up before users can take calls
- Created Training list view
- Created training detail view

### CODE REVIEW

**Comment by BradenM** [comment]
@JamesRArroyo mind rebasing this? hmu if anything questionable comes up. thanks

**Comment by JamesRArroyo** [comment]
@BradenM . Code is rebased

File: src/components/phone/ContactCard.vue [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    </div>
+    <!--- info card --->
+    <div class="info-card">
+      <more-info v-if="toggleOpen" />
+    </div>
+    <!--- Buttons --->
+    <div class="flex flex-row justify-around items-center m-3">
+      <div class="flex-col">
+        <base-button
+          v-if="completedTraining"
```
Comment (BradenM) [inline src/components/phone/ContactCard.vue]: Hate to do this to you... but this component no longer exists :man_shrugging: 

As of 7968b72 I made a lot of changes to it and moved it to `src/components/phone/AgentCard.vue`

**Comment by JamesRArroyo** [comment]
Moved training modal implementation over to the new agent card

**Pull request merged**